src/NpmUtilities.js

```diff
@@ -18,22 +18,29 @@ export default class NpmUtilities {
       stdio: ["ignore", "pipe", "pipe"],
     };
 
+    if (registry) {
+      opts.env = {npm_config_registry: registry};
+    }
+
     ChildProcessUtilities.spawn("npm", args, opts, callback);
   }
 
   @logger.logifySync()
-  static addDistTag(packageName, version, tag) {
-    ChildProcessUtilities.execSync(`npm dist-tag add ${packageName}@${version} ${tag}`);
+  static addDistTag(packageName, version, tag, registry) {
+    const opts = NpmUtilities.getTagOpts(registry);
+    ChildProcessUtilities.execSync(`npm dist-tag add ${packageName}@${version} ${tag}`, opts);
```

Comment:
making this part of the public api like this makes me think you're shelling out to npm every time.  I feel like 

```
ChildProcessUtilities.execSync(`npm dist-tag add ${packageName}@${version} ${tag}`, {registry});
```

would be better, as would making getTagOpts a function

```
ChildProcessUtilities.execSync(`npm dist-tag add ${packageName}@${version} ${tag}`, getTagOpts(registry));
```